kpatch-build/create-diff-object.c

```diff
@@ -2361,8 +2378,9 @@ static void kpatch_create_intermediate_sections(struct kpatch_elf *kelf,
 		    !strcmp(sec->name, ".rela.kpatch.dynrelas"))
 			continue;
 		list_for_each_entry_safe(rela, safe, &sec->relas, list) {
-			if (rela->sym->sec)
+			if (!may_need_dynrela(rela))
```

Comment:
This would be more readable if it were restructured slightly:
```
if (rela->sym->sec && !function_ptr_rela(rela))
```
In other words, keep the original check, but put the new check in a dedicated `function_ptr_rela()` function.

And also `function_ptr_rela()` needs some comments, specifically a) why we do dynrelas for function pointers; and b) why functions with '.' are skipped.

Reply:
OK, will do it.